hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/serialization/Value.java

```diff
@@ -94,4 +95,23 @@ public int getClassId() {
         return HibernateDataSerializerHook.VALUE;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        Value value1 = (Value) o;
+        return timestamp == value1.timestamp && Objects.equals(value, value1.value);
```

Comment:
I think `timestamp` should be actually excluded from `equals/hashcode` because what actually matters are pairs _(version, object)_. @enozcan what do you think?

Reply:
Could be. But doesn't it break the atomicity of map#replace spin?